ocaml/xapi/dbsync.ml

```diff
@@ -71,7 +71,16 @@ let update_env () =
     )
 
 let setup () =
+
+  let cache_xapi_globs () =
+    Server_helpers.exec_with_new_task "cache_xapi_globs" ~task_in_database:false
+    (fun __context ->
+      if !Xapi_globs.localhost_ref = Ref.null then
+        Xapi_globs.localhost_ref := Helpers.get_localhost ~__context)
+  in
+
   try
+    cache_xapi_globs ();
```

Comment:
realised this morning that i don't know whether `task1` blocks `task2` or not:

```
Server_helpers.exec_with_new_task "task1" (fun _context -> ...);
Server_helpers.exec_with_new_task "task2" (fun _context -> ...);
```

if not, then I need to change this code

Reply:
I think that all we need is `Server_helpers.exec_with_new_task "set localhost_ref" (fun __context -> Xapi_globs.localhost_ref := Helpers.get_localhost ~__context)` immediately under `let update_env () =`. And delete the corresponding line in dbsync_slave.ml.

Reply:
so `exec_with_new_task` does block?